cwms-data-api/src/main/java/cwms/cda/datasource/DirectUserPreparer.java

```diff
@@ -14,15 +18,23 @@ public DirectUserPreparer(String user) {
 
     @Override
     public Connection prepare(Connection conn) {
-        if (user != null) {
-            String sql = "begin cwms_env.set_session_user_direct(upper(?)); end;";
-            try (PreparedStatement setApiUser = conn.prepareStatement(sql)) {
-                setApiUser.setString(1,user);
-                setApiUser.execute();                
-            } catch (Exception e) {
-                throw new DataAccessException("Unable to set user session.  "
-                        + "user empty = " + user.isEmpty(), e);
+        try {
```

Comment:
well, it might work. presumably if the database is in read only mode the oracle jdbc driver would actually respond as such.